### PR TITLE
ANN: Add support for E0536 error

### PR DESCRIPTION
**Pull request by missingdays**
changelog: Add support for E0536 error

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/fixes/ConvertCfgNotMultipleArgsToMultipleNotsFix.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        val factory = RsPsiFactory(project)
+        val newItem = factory.createMetaItem(buildString {
+            append("all(")
+
+            append(metaItemList.joinToString(", ") {
+                "not(${it.text})"
+            })
+
+            append(")")
+        })
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/fixes/ConvertCfgNotMultipleArgsToMultipleNotsFix.kt:49]: It seems like it's possible to extract this code and unify it with the code in `fixText`

Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/fixes/ConvertCfgNotMultipleArgsToMultipleNotsFix.kt:49]:
```suggestion
        val newItem = factory.createMetaItem(metaItemList.joinToString(prefix = "all(", separator = ", ", postfix = ")") {
                "not(${it.text})"
            })
```

File: src/main/kotlin/org/rust/ide/fixes/ConvertCfgNotMultipleArgsToMultipleNotsFix.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        "Convert to `all($negatedArguments)`"
+    }
+
+    override fun getFamilyName(): String = "Fix `not` cfg-pattern"
+    override fun getText(): String = fixText
+
+    override fun isAvailable(project: Project, file: PsiFile, startElement: PsiElement, endElement: PsiElement): Boolean {
+        val metaItem = startElement as? RsMetaItem ?: return false
+        return metaItem.metaItemArgsList.size > 1
+    }
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/fixes/ConvertCfgNotMultipleArgsToMultipleNotsFix.kt:34]: We usually don't use `isAvailable` in quick fixes and assume that the fix is available if it was supplied by an annotator/inspection.

File: src/main/kotlin/org/rust/ide/fixes/ConvertCfgNotMultipleArgsToMultipleNotsFix.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        if (metaItemArgsList.size < 2) {
+            return@run ""
+        }
+
+        val negatedArguments = metaItemArgsList.joinToString(", ") { "not(${it.text})" }
+
+        "Convert to `all($negatedArguments)`"
+    }
+
+    override fun getFamilyName(): String = "Fix `not` cfg-pattern"
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/fixes/ConvertCfgNotMultipleArgsToMultipleNotsFix.kt:28]:
```suggestion
    override fun getFamilyName(): String = "Convert `not(a, b)` cfg-pattern to `all(not(a), not(b))`"
```

File: src/main/kotlin/org/rust/ide/fixes/ConvertCfgNotMultipleArgsToMultipleNotsFix.kt (hunk 2 above)
Comment (missingdays) [reply src/main/kotlin/org/rust/ide/fixes/ConvertCfgNotMultipleArgsToMultipleNotsFix.kt:34]: The problem is that the fix doesn't work if there are zero arguments passed to `not()`. But it's still an error to do so
Would you suggest to instead do this check on the inspection level and not create the fix at all?

Comment (vlad20012) [reply src/main/kotlin/org/rust/ide/fixes/ConvertCfgNotMultipleArgsToMultipleNotsFix.kt:34]: We usually use `Fix.createIfCompatible` pattern. See for example `ChangeReprAttributeFix.createIfCompatible`

**Review by vlad20012 — APPROVED** [review]
bors r+

**Pull request merged**